From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tiopex <tiopxyz@gmail.com>
Date: Sun, 9 Aug 2026 09:11:00 +0200
Subject: drm/msm/dsi: reparent byte/pixel src clocks to XO on disable

> X-Git-Archeology: - Revision cd3cfe78bd00658dce3d89b584e9c1e2554ffb65: https://github.com/armbian/build/commit/cd3cfe78bd00658dce3d89b584e9c1e2554ffb65
> X-Git-Archeology:   Date: Sun, 09 Aug 2026 09:11:00 +0200
> X-Git-Archeology:   From: Alex Ling <ling_kasim@hotmail.com>
> X-Git-Archeology:   Subject: sm8750: Import kernel 7.1 patches from ROCKNIX
> X-Git-Archeology:
---
 drivers/gpu/drm/msm/dsi/dsi_host.c | 12 ++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/msm/dsi/dsi_host.c b/drivers/gpu/drm/msm/dsi/dsi_host.c
index 111111111111..222222222222 100644
--- a/drivers/gpu/drm/msm/dsi/dsi_host.c
+++ b/drivers/gpu/drm/msm/dsi/dsi_host.c
@@ -127,6 +127,8 @@ struct msm_dsi_host {
 	struct clk *pixel_src_clk;
 	struct clk *dsi_pll_byte_clk;
 	struct clk *dsi_pll_pixel_clk;
+	struct clk *byte_src_parent;
+	struct clk *pixel_src_parent;
 
 	unsigned long byte_clk_rate;
 	unsigned long byte_intf_clk_rate;
@@ -416,6 +418,11 @@ int dsi_link_clk_set_rate_6g_v2_9(struct msm_dsi_host *msm_host)
 	struct device *dev = &msm_host->pdev->dev;
 	int ret;
 
+	if (!msm_host->byte_src_parent)
+		msm_host->byte_src_parent = clk_get_parent(msm_host->byte_src_clk);
+	if (!msm_host->pixel_src_parent)
+		msm_host->pixel_src_parent = clk_get_parent(msm_host->pixel_src_clk);
+
 	/*
 	 * DSI PHY PLLs have to be enabled to allow reparenting to them, so
 	 * cannot use assigned-clock-parents.
@@ -2556,6 +2563,11 @@ int msm_dsi_host_power_off(struct mipi_dsi_host *host)
 
 	pinctrl_pm_select_sleep_state(&msm_host->pdev->dev);
 
+	if (msm_host->byte_src_parent)
+		clk_set_parent(msm_host->byte_src_clk, msm_host->byte_src_parent);
+	if (msm_host->pixel_src_parent)
+		clk_set_parent(msm_host->pixel_src_clk, msm_host->pixel_src_parent);
+
 	cfg_hnd->ops->link_clk_disable(msm_host);
 	pm_runtime_put(&msm_host->pdev->dev);
 
-- 
Armbian

